addons/pos_online_payment/static/src/app/utils/order_payment_validation.js

```diff
@@ -46,6 +46,12 @@ patch(OrderPaymentValidation.prototype, {
     },
     //@override
     async isOrderValid(isForceValidate) {
+        if (
+            this.order.refunded_order_id?.pos_reference.includes("Self-Order") &&
```

Comment:
Pos reference will not include "self-order" anymore, you need to check "source" fields on order